ttnn/cpp/ttnn/operations/pool/generic/device/pool_multi_core_program_factory.cpp

```diff
@@ -367,21 +410,22 @@ Pool2D::MultiCore::cached_program_t pool2d_multi_core_sharded_with_halo_v2_impl_
         in_nblocks_c,
         max_rows_for_reduction};
 
-    auto reduce_op = tt::tt_metal::ReduceOpMath::MAX;
-    auto reduce_dim = tt::tt_metal::ReduceOpDim::H;
+    const auto reduce_op = get_reduce_op(pool_type);
+    const auto reduce_dim = tt::tt_metal::ReduceOpDim::H;
+    const auto reduce_op_defines = reduce_op_utils::get_defines(reduce_op, reduce_dim);
     auto compute_config = ComputeConfig{
         .math_fidelity = MathFidelity::HiFi4,
         .fp32_dest_acc_en = false,
```

Comment:
Depending on the size of the inputs, reducing on sum will give bad PCCs unless fp32 dest acc is enabled or you use an algorithm to reduce error (tree/pairwise summation  or a variation of Kahan: https://en.wikipedia.org/wiki/Kahan_summation_algorithm).

See https://github.com/tenstorrent/tt-metal/issues/13621 for details. Now reduce tries to use fp32 dest acc. You probably want to do the same, at least for sum.

Reply:
I see. I tried to enable fp32_dest_acc but IIUC it will reduce the available DST tiles from 8 to 4 according to

https://github.com/tenstorrent/tt-metal/blob/5d0170e0a60435f88b4d2b0e86472e4df5d457c8/tech_reports/matrix_engine/matrix_engine.md?plain=1#L76-L81

This will put more use cases into wide reduction (channel >= 256), which doesn't work currently due to #14459. Since it will need more modify and test, also currently the sizes of inputs are still in tolerance, I feel it would be better to defer this change into a follow-up PR. WDYT?

Reply:
@jerrysky3  Re: fp32_dest_acc, yes, this can be deferred. It's good to keep in mind if your PCC is too low.